### ISSUE TITLE
Rails server not booting with gem enabled

### ISSUE THREAD

**Issue by matteo-rossi-wise** [yabeda-rb/yabeda-rails#10]
Hi everyone, I'm experiencing this weird issue while trying to boot my Rails app:

Ruby: `2.7.0`
Rails: `6.0.2.2`

---

```shell
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: docstring)
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/prometheus-client-2.0.0/lib/prometheus/client/registry.rb:56:in `gauge'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-prometheus-0.1.5/lib/yabeda/prometheus/adapter.rb:23:in `register_gauge!'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-0.5.0/lib/yabeda/base_adapter.rb:9:in `register!'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-0.5.0/lib/yabeda/dsl/class_methods.rb:71:in `block in register_metric'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-0.5.0/lib/yabeda/dsl/class_methods.rb:71:in `each_value'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-0.5.0/lib/yabeda/dsl/class_methods.rb:71:in `register_metric'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-0.5.0/lib/yabeda/dsl/class_methods.rb:48:in `gauge'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-puma-plugin-0.3.0/lib/puma/plugin/yabeda.rb:18:in `block in start'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-0.5.0/lib/yabeda.rb:68:in `class_eval'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-0.5.0/lib/yabeda.rb:68:in `block in configure!'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-0.5.0/lib/yabeda.rb:66:in `each'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-0.5.0/lib/yabeda.rb:66:in `configure!'
  ../.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/yabeda-rails-0.5.0/lib/yabeda/rails/railtie.rb:20:in `block in <class:Railtie>'
```

This is my current setup:

**Gemfile**

```ruby
# Prometheus instrumentation library for Ruby applications
gem "prometheus-client"

# Collect and export basic metrics for Rails applications
gem "yabeda-rails"

# Collects Puma web-server metrics from puma control panel
gem "yabeda-puma-plugin"

# Prometheus integration to export evil metrics
gem "yabeda-prometheus"
```

**Gemfile.lock**

```ruby
prometheus-client (2.0.0)
yabeda (0.5.0)
      concurrent-ruby
      dry-initializer
    yabeda-prometheus (0.1.5)
      yabeda
    yabeda-puma-plugin (0.3.0)
      json
      puma
      yabeda (~> 0.2)
    yabeda-rails (0.5.0)
      rails
      yabeda (~> 0.4)
```

**config.ru**

```ruby
require_relative "config/environment"

# Insert _before_ the main application:
use Yabeda::Prometheus::Exporter

run Rails.application
```

**config/puma.rb**

```ruby
# Allow Yabeda to collect puma metrics
#
activate_control_app
plugin :yabeda
```

By removing `yabeda-rails`, with the current setup, it's working as expected and the app boots.

**Comment by Envek** [comment]
Either upgrade `yabeda-prometheus` gem to version 0.2 or newer or downgrade `prometheus-client` to version 0.8.

`prometheus-client` changed their API in 1.0 so `yabeda-prometheus` switched to it in version `0.2` and dropped support for older prometheus-client versions.

See https://github.com/yabeda-rb/yabeda-prometheus/blob/master/CHANGELOG.md#020---2020-01-14 for details

**Comment by Envek** [comment]
gem `yabeda-rails` is just automatically executes `Yabeda.configure!` for you which makes `yabeda-prometheus` to push metrics configuration to `prometheus-client`.

By removing `yabeda-rails` you essentially disable Yabeda altogether.

**Comment by matteo-rossi-wise** [comment]
Thanks for your support, now it's working as expected 🚀 
We can close this issue.

The outcome:

```ruby
prometheus-client (1.0.0)
yabeda (0.5.0)
  concurrent-ruby
  dry-initializer
yabeda-prometheus (0.5.0)
  prometheus-client (~> 1.0)
  yabeda (~> 0.5)
yabeda-puma-plugin (0.3.0)
  json
  puma
  yabeda (~> 0.2)
yabeda-rails (0.5.0)
  rails
  yabeda (~> 0.4)
```

**Comment by Envek** [comment]
Also published [yabeda-prometheus 0.6.0](https://github.com/yabeda-rb/yabeda-prometheus/releases/tag/v0.6.0) so you can use prometheus-client 2.0.0
